Failing msstatistics tests

I get the following output from `test_using_tasks.py`:
```sh
$ pytest -s pyraf/tests/test_using_tasks.py
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.7.9, pytest-6.1.1, py-1.9.0, pluggy-0.13.1
DRAGONS_TEST directory: None
rootdir: /home/jturner/repos/pyraf, configfile: setup.cfg
plugins: hypothesis-6.8.5, arraydiff-0.3, astropy-header-0.1.2, doctestplus-0.0.0, openfiles-0.5.0, remotedata-0.3.2, asdf-2.8.0.dev224+gba09e57.d20210407, dragons-3.1.0.dev0
collecting ... 

      +------------------------------------------------------------+
      |       Space Telescope Science Data Analysis System         |
      |                   STSDAS Version 3.18.3                    |
      |                                                            |
      |   Space Telescope Science Institute, Baltimore, Maryland   |
      |   Copyright (C) 2014 Association of Universities for       |
      |            Research in Astronomy, Inc.(AURA)               |
      |       See stsdas$copyright.stsdas for terms of use.        |
      |         For help, send e-mail to help@stsci.edu            |
      |                                                            |
      +------------------------------------------------------------+
collected 7 items                                                                                                                                                                                                 

pyraf/tests/test_using_tasks.py         (npix = 0)              >Number of good pixels
         (min = 0.0)            >Minimum pixel value
         (max = 0.0)            >Maximum pixel value
         (sum = 0.0)            >Sum
        (mean = 0.0)            >Mean
      (stddev = 0.0)            >Standard deviation
   (histmidpt = 0.0)            >Median
    (histpeak = 0.0)            >Mode
        (skew = 0.0)            >Skewness
        (kurt = 0.0)            >Kurtosis
       (wmean = 0.0)            >Weighted mean
        (wvar = 0.0)            >Weighted variance
        (mode = "al")           
.        (npix = 0)              >Number of good pixels
         (min = 0.0)            >Minimum pixel value
         (max = 0.0)            >Maximum pixel value
         (sum = 0.0)            >Sum
        (mean = 0.0)            >Mean
      (stddev = 0.0)            >Standard deviation
   (histmidpt = 0.0)            >Median
    (histpeak = 0.0)            >Mode
        (skew = 0.0)            >Skewness
        (kurt = 0.0)            >Kurtosis
       (wmean = 0.0)            >Weighted mean
        (wvar = 0.0)            >Weighted variance
        (mode = "al")           
.        (npix = 0)              >Number of good pixels
         (min = 0.0)            >Minimum pixel value
         (max = 0.0)            >Maximum pixel value
         (sum = 0.0)            >Sum
        (mean = 0.0)            >Mean
      (stddev = 0.0)            >Standard deviation
   (histmidpt = 0.0)            >Median
    (histpeak = 0.0)            >Mode
        (skew = 0.0)            >Skewness
        (kurt = 0.0)            >Kurtosis
       (wmean = 0.0)            >Weighted mean
        (wvar = 0.0)            >Weighted variance
        (mode = "al")           
# image                         npix        mean      stddev
#
/home/jturner/repos/pyraf     16384.     1159.94     23.3568 
/home/jturner/repos/pyraf     16384.     1159.57      37.312 
        (npix = 16384)          >Number of good pixels
         (min = 1141.0)         >Minimum pixel value
         (max = 4065.0)         >Maximum pixel value
         (sum = 18998468.0)     >Sum
        (mean = 1159.57446289)  >Mean
      (stddev = 37.3120422363)  >Standard deviation
   (histmidpt = 0.0)            >Median
    (histpeak = 0.0)            >Mode
        (skew = 0.0)            >Skewness
        (kurt = 0.0)            >Kurtosis
       (wmean = 0.0)            >Weighted mean
        (wvar = 0.0)            >Weighted variance
        (mode = "al")           
F        (npix = 0)              >Number of good pixels
         (min = 0.0)            >Minimum pixel value
         (max = 0.0)            >Maximum pixel value
         (sum = 0.0)            >Sum
        (mean = 0.0)            >Mean
      (stddev = 0.0)            >Standard deviation
   (histmidpt = 0.0)            >Median
    (histpeak = 0.0)            >Mode
        (skew = 0.0)            >Skewness
        (kurt = 0.0)            >Kurtosis
       (wmean = 0.0)            >Weighted mean
        (wvar = 0.0)            >Weighted variance
        (mode = "al")           
# image                         npix        mean      stddev
#
/home/jturner/repos/pyraf     16384.     1159.94     23.3568 
/home/jturner/repos/pyraf     16384.     1159.57      37.312 
        (npix = 0)              >Number of good pixels
         (min = 0.0)            >Minimum pixel value
         (max = 0.0)            >Maximum pixel value
         (sum = 0.0)            >Sum
        (mean = 0.0)            >Mean
      (stddev = 0.0)            >Standard deviation
   (histmidpt = 0.0)            >Median
    (histpeak = 0.0)            >Mode
        (skew = 0.0)            >Skewness
        (kurt = 0.0)            >Kurtosis
       (wmean = 0.0)            >Weighted mean
        (wvar = 0.0)            >Weighted variance
        (mode = "al")           
.        (npix = 0)              >Number of good pixels
         (min = 0.0)            >Minimum pixel value
         (max = 0.0)            >Maximum pixel value
         (sum = 0.0)            >Sum
        (mean = 0.0)            >Mean
      (stddev = 0.0)            >Standard deviation
   (histmidpt = 0.0)            >Median
    (histpeak = 0.0)            >Mode
        (skew = 0.0)            >Skewness
        (kurt = 0.0)            >Kurtosis
       (wmean = 0.0)            >Weighted mean
        (wvar = 0.0)            >Weighted variance
        (mode = "al")           
# image                         npix        mean      stddev
#
/home/jturner/repos/pyraf     16384.     1159.94     23.3568 
/home/jturner/repos/pyraf     16384.     1159.57      37.312 
        (npix = 16384)          >Number of good pixels
         (min = 1141.0)         >Minimum pixel value
         (max = 4065.0)         >Maximum pixel value
         (sum = 18998468.0)     >Sum
        (mean = 1159.57446289)  >Mean
      (stddev = 37.3120422363)  >Standard deviation
   (histmidpt = 0.0)            >Median
    (histpeak = 0.0)            >Mode
        (skew = 0.0)            >Skewness
        (kurt = 0.0)            >Kurtosis
       (wmean = 0.0)            >Weighted mean
        (wvar = 0.0)            >Weighted variance
        (mode = "al")           
F        (npix = 0)              >Number of good pixels
         (min = 0.0)            >Minimum pixel value
         (max = 0.0)            >Maximum pixel value
         (sum = 0.0)            >Sum
        (mean = 0.0)            >Mean
      (stddev = 0.0)            >Standard deviation
   (histmidpt = 0.0)            >Median
    (histpeak = 0.0)            >Mode
        (skew = 0.0)            >Skewness
        (kurt = 0.0)            >Kurtosis
       (wmean = 0.0)            >Weighted mean
        (wvar = 0.0)            >Weighted variance
        (mode = "al")           
# image                         npix        mean      stddev
#
dev$pix                      262144.     108.315     131.298 
        (npix = 262144)         >Number of good pixels
         (min = -1.0)           >Minimum pixel value
         (max = 19936.0)        >Maximum pixel value
         (sum = 28394234.0)     >Sum
        (mean = 108.315406799)  >Mean
      (stddev = 131.298019409)  >Standard deviation
   (histmidpt = 0.0)            >Median
    (histpeak = 0.0)            >Mode
        (skew = 0.0)            >Skewness
        (kurt = 0.0)            >Kurtosis
       (wmean = 0.0)            >Weighted mean
        (wvar = 0.0)            >Weighted variance
        (mode = "al")           
        (npix = 0)              >Number of good pixels
         (min = 0.0)            >Minimum pixel value
         (max = 0.0)            >Maximum pixel value
         (sum = 0.0)            >Sum
        (mean = 0.0)            >Mean
      (stddev = 0.0)            >Standard deviation
   (histmidpt = 0.0)            >Median
    (histpeak = 0.0)            >Mode
        (skew = 0.0)            >Skewness
        (kurt = 0.0)            >Kurtosis
       (wmean = 0.0)            >Weighted mean
        (wvar = 0.0)            >Weighted variance
        (mode = "al")           
        (npix = 0)              >Number of good pixels
         (min = 0.0)            >Minimum pixel value
         (max = 0.0)            >Maximum pixel value
         (sum = 0.0)            >Sum
        (mean = 0.0)            >Mean
      (stddev = 0.0)            >Standard deviation
   (histmidpt = 0.0)            >Median
    (histpeak = 0.0)            >Mode
        (skew = 0.0)            >Skewness
        (kurt = 0.0)            >Kurtosis
       (wmean = 0.0)            >Weighted mean
        (wvar = 0.0)            >Weighted variance
        (mode = "al")           
# image                         npix        mean      stddev
#
dev$pix                      262144.     108.315     131.298 
        (npix = 262144)         >Number of good pixels
         (min = -1.0)           >Minimum pixel value
         (max = 19936.0)        >Maximum pixel value
         (sum = 28394234.0)     >Sum
        (mean = 108.315406799)  >Mean
      (stddev = 131.298019409)  >Standard deviation
   (histmidpt = 0.0)            >Median
    (histpeak = 0.0)            >Mode
        (skew = 0.0)            >Skewness
        (kurt = 0.0)            >Kurtosis
       (wmean = 0.0)            >Weighted mean
        (wvar = 0.0)            >Weighted variance
        (mode = "al")           
.dev$pix -> /tmp/pytest-of-jturner/pytest-10/test_dqbits_mscombine0/dqbits_im1.fits
dev$pix -> /tmp/pytest-of-jturner/pytest-10/test_dqbits_mscombine0/dqbits_im2.fits
        (bit1 = no)             Reject pixels with 
        (bit2 = no)             bit set ?
        (bit3 = no)             
        (bit4 = no)             
        (bit5 = no)             
        (bit6 = no)             
        (bit7 = no)             
        (bit8 = no)             
        (bit9 = no)             
       (bit10 = no)             
       (bit11 = no)             
       (bit12 = no)             
       (bit13 = no)             
       (bit14 = no)             
       (bit15 = no)             
       (bit16 = no)             
        (mode = "al")           
        (bit1 = no)             Reject pixels with 
        (bit2 = yes)            bit set ?
        (bit3 = no)             
        (bit4 = yes)            
        (bit5 = no)             
        (bit6 = yes)            
        (bit7 = no)             
        (bit8 = yes)            
        (bit9 = no)             
       (bit10 = no)             
       (bit11 = no)             
       (bit12 = no)             
       (bit13 = no)             
       (bit14 = no)             
       (bit15 = no)             
       (bit16 = no)             
        (mode = "al")           
MSCOMBINE Version 14Mar07 at Thu 13:13:14 17-Mar-2022
Warning: Image header parameter not found (INSTRUME)
Warning: Image header parameter not found (NEXTEND)
Ending MSCOMBINE at Thu 13:13:14 17-Mar-2022
        (bit1 = no)             Reject pixels with 
        (bit2 = yes)            bit set ?
        (bit3 = no)             
        (bit4 = yes)            
        (bit5 = no)             
        (bit6 = yes)            
        (bit7 = no)             
        (bit8 = yes)            
        (bit9 = no)             
       (bit10 = no)             
       (bit11 = no)             
       (bit12 = no)             
       (bit13 = no)             
       (bit14 = no)             
       (bit15 = no)             
       (bit16 = no)             
        (mode = "al")           
.

==================================================================================================== FAILURES =====================================================================================================
______________________________________________________________________________________ test_pset_msstatistics_science_array _______________________________________________________________________________________

_iraf_pset_init = None
_data = {'dqbits': {'input1': '/tmp/pytest-of-jturner/pytest-10/test_pset_msstatistics_science0/dqbits_im1.fits', 'input2': '/...cs_science0/dqbits_out.fits'}, 'pset': {'input1': '/home/jturner/repos/pyraf/pyraf/tests/data/pset_msstat_input.fits'}}

    def test_pset_msstatistics_science_array(_iraf_pset_init, _data):
        """Expect decent data
        """
        # Run msstat, which sets egstp values.  Test PSET par passing to task
        # command function as a task top-level par passing msstat.nsstatpar.arrays
        # and msstat.nsstatpar.clarray, as if they were just msstat pars...
        #     arrays='science' (check "science" arrays only)
        #    clarray='science' (return data to egstp from final "error" array)
        # So, expect vals from second (final) science array.
    
        iraf.msstatistics(_data['pset']['input1'],
                          arrays='science',
                          clarray='science')
        iraf.egstp.lParam()
    
>       assert iraf.egstp.npix == 277704, str(iraf.egstp.npix)
E       AssertionError: 16384
E       assert 16384 == 277704
E        +  where 16384 = <IrafPset egstp (msstatistics$egstp.par) Pkg: mstools Bin: stsdasbin$>.npix
E        +    where <IrafPset egstp (msstatistics$egstp.par) Pkg: mstools Bin: stsdasbin$> = iraf.egstp

pyraf/tests/test_using_tasks.py:197: AssertionError
___________________________________________________________________________________________ test_pset_msstatistics_191 ____________________________________________________________________________________________

_iraf_pset_init = None
_data = {'dqbits': {'input1': '/tmp/pytest-of-jturner/pytest-10/test_pset_msstatistics_1910/dqbits_im1.fits', 'input2': '/tmp/...istics_1910/dqbits_out.fits'}, 'pset': {'input1': '/home/jturner/repos/pyraf/pyraf/tests/data/pset_msstat_input.fits'}}

    def test_pset_msstatistics_191(_iraf_pset_init, _data):
        """Expect egstp to be properly cleared and used again
        NOTE: Referenced issue, 191, is no longer available (trac).
        """
        # Regression test to make sure a task can be sent an unadorned PSET
        # par as a regular function argument (without scope/PSET name given).
        # This regression-tests #191.
    
        # repeat first call to msstat, verify correct results and that we did
        # not hit anything which exists due to previous calls
        iraf.msstat(_data['pset']['input1'], arrays='science', clarray='science')
        iraf.egstp.lParam()
    
>       assert iraf.egstp.npix == 277704, str(iraf.egstp.npix)
E       AssertionError: 16384
E       assert 16384 == 277704
E        +  where 16384 = <IrafPset egstp (msstatistics$egstp.par) Pkg: mstools Bin: stsdasbin$>.npix
E        +    where <IrafPset egstp (msstatistics$egstp.par) Pkg: mstools Bin: stsdasbin$> = iraf.egstp

pyraf/tests/test_using_tasks.py:234: AssertionError
============================================================================================= short test summary info =============================================================================================
FAILED pyraf/tests/test_using_tasks.py::test_pset_msstatistics_science_array - AssertionError: 16384
FAILED pyraf/tests/test_using_tasks.py::test_pset_msstatistics_191 - AssertionError: 16384
=========================================================================================== 2 failed, 5 passed in 0.74s ===========================================================================================
```
Maybe I'm misunderstanding some of the fiddly indirection here, but it looks like the input file `pset_msstat_input.fits` has 2 `SCI` extensions with dimensions of 128x128 = 16384, so where does `npix == 277704` in the test definition come from?

I suspect that part of the issue here may be that you're skipping these tests because STSDAS hasn't been ported to 64 bits?

This is not causing me any specific problem; I'm just reporting the failure.

--- a/pyraf/tests/test_using_tasks.py
+++ b/pyraf/tests/test_using_tasks.py
@@ -194,10 +194,10 @@ def test_pset_msstatistics_science_array(_iraf_pset_init, _data):
                       clarray='science')
     iraf.egstp.lParam()
 
-    assert iraf.egstp.npix == 277704, str(iraf.egstp.npix)
-    _assertApproxEqual(iraf.egstp.min, 1116.0)
-    _assertApproxEqual(iraf.egstp.max, 14022.0)
-    _assertApproxEqual(iraf.egstp.sum, 321415936.0)
+    assert iraf.egstp.npix == 16384, str(iraf.egstp.npix)
+    _assertApproxEqual(iraf.egstp.min, 1141.0)
+    _assertApproxEqual(iraf.egstp.max, 4065.0)
+    _assertApproxEqual(iraf.egstp.sum, 18998468.0)
 
 
 def test_pset_msstatistics_zeroed_error_array(_iraf_pset_init, _data):
@@ -231,10 +231,10 @@ def test_pset_msstatistics_191(_iraf_pset_init, _data):
     iraf.msstat(_data['pset']['input1'], arrays='science', clarray='science')
     iraf.egstp.lParam()
 
-    assert iraf.egstp.npix == 277704, str(iraf.egstp.npix)
-    _assertApproxEqual(iraf.egstp.min, 1116.0)
-    _assertApproxEqual(iraf.egstp.max, 14022.0)
-    _assertApproxEqual(iraf.egstp.sum, 321415936.0)
+    assert iraf.egstp.npix == 16384, str(iraf.egstp.npix)
+    _assertApproxEqual(iraf.egstp.min, 1141.0)
+    _assertApproxEqual(iraf.egstp.max, 4065.0)
+    _assertApproxEqual(iraf.egstp.sum, 18998468.0)
 
     _unlearn_egstp(iraf.egstp)
     iraf.msstatistics(_data['pset']['input1'],
@@ -251,10 +251,10 @@ def test_pset_msstatistics_191(_iraf_pset_init, _data):
     iraf.msstat(_data['pset']['input1'], arrays='science', clarray='science')
     iraf.egstp.lParam()
 
-    assert iraf.egstp.npix == 277704, str(iraf.egstp.npix)
-    _assertApproxEqual(iraf.egstp.min, 1116.0)
-    _assertApproxEqual(iraf.egstp.max, 14022.0)
-    _assertApproxEqual(iraf.egstp.sum, 321415936.0)
+    assert iraf.egstp.npix == 16384, str(iraf.egstp.npix)
+    _assertApproxEqual(iraf.egstp.min, 1141.0)
+    _assertApproxEqual(iraf.egstp.max, 4065.0)
+    _assertApproxEqual(iraf.egstp.sum, 18998468.0)
 
 
 def test_pset_msstatistics_save_data(_iraf_pset_init, _data):

